test/blackbox-tests/test-cases/cram/dune

```diff
@@ -0,0 +1,5 @@
+(cram)
+
+(cram
+ (enabled_if %{bin-available:zsh})
```

Comment:
If you do this, you will have to add zsh to the CI so we can run the tests. Probably best if you restrict to unix too.

Reply:
I think the macos CI machine should zsh available.

What's the additional benefits to add restriction to unix in addition to checking whether zsh exists?

Reply:
Well actually I'm thinking that we should be using `bin-available`. If we miss putting `zsh` in the CI then this test will never be run, and it might surprise some other devs when they have `zsh` if this test is broken. So we have two choices:

1. Either we make `zsh` mandatory (hence the unix restriction) so that the test is always seen to run.
2. We make a dummy `zsh` or any other shell, just to check how it works. (echo would probably work)
3. We don't test zsh.

Reply:
I see. I did have similar feeling when I added the test, but I ended up with adding them since I wanted to have some non-artificial tests.

But given that your concern is solid, I am in favor of doing (1) or (3) as you suggested. I don't think (2) is a good option since it seems that dune's cram test executor expects the shell at least have (a) output redirecting and (b) script sourcing features and making a convincing dummy `zsh` above the level of what is already done in `shell-opt.t` requires some significant efforts.

Regarding (1), I'm in general in favor of it, but since it involves changing the CI configuration, I don't think I should make the call.

If the maintainers of dune is not in favor of (1), then I think we should take (3) as missing this test case isn't crucial for the completeness of testing the feature added in this PR.

Reply:
(thank you for taking a look btw!)

Reply:
How should we reach to a consensus here?